### PR TITLE
Make multiple scenarios backwards compatible

### LINKED ISSUE #982: Decouple Kids and GobstonesJr code
See https://github.com/mumuki/mumuki-laboratory/pull/955, there are many places where the kids layout is assuming GobstonesJr code.  Also, there is high coupling between custom editors and kids layouts, but they are different things. This must be avoided by all means. 

We should 

- [x] document the API exposed by laboratory, composed of css classes, js objects and css selectors.  
- [x] ensure Gobstones Runner is using laboratory through this well-defined API
- [x] allow runners to query the current laboratory version, in order to check if they are running againts a valid version.  They could at least print a warning in the console if it is not the case, to help debugging. 
- [x] define which JS version are we going to target
- [x] reinforce `_` for private methods convention

### PR DESCRIPTION
**Pull request by luchotc**
This changes allows multiple scenarios to work well with the current version of gobstones.
~This should be reverted after deploying the new version of it~

Fixes #982

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Related to https://github.com/mumuki/mumuki-laboratory/issues/982

I am not sure we have to actually rollback this change, but instead allow the runner to override this behaviour. Let's discuss it further

**Comment by flbulgarelli** [comment]
Related to https://github.com/mumuki/mumuki-laboratory/issues/982

**Comment by flbulgarelli** [comment]
:up: I did some refactors: 

* Removed deprecated code
* Prefixes private API with `_`
* Documented the actual API
* Added more consistent, backward compatible selectors

Thus, I think this fixes #982

File: app/assets/javascripts/application/kids.js [hunk 1]
```diff
@@ -80,151 +54,189 @@ mumuki.load(function () {
...
       return $('#kids-results');
     },
 
-    getResultsAbortedModal: function () {
+    _getResultsAbortedModal: function () {
       return $('#kids-results-aborted');
     },
 
-    getCharaterImage: function () {
+    _getCharaterImage: function () {
```
Comment (luchotc) [inline app/assets/javascripts/application/kids.js]: typo! tbf, it was already there :stuck_out_tongue:

**Comment by luchotc** [comment]
Nice @flbulgarelli! :clap:

**Pull request merged**